### DISCUSSION TITLE
Magnet with Multipoles

### DISCUSSION THREAD

**Discussion by oscarxblanco** [atcollab/at#618] [Python AT]
Dear all,
I am trying to add multipolar components to a sextupole. Is there any suggestion on how to do it on pyat ?

I tried increasing the length of the polynoms and setting MaxOrder to a value bigger than 3. 
However, an error appears when trying to change the MaxOrder value : 
```
  File "/home/sources/physmach/blanco-garcia/codes/pyenv/venv_pyatmaster/lib/python3.8/site-packages/at/lattice/elements.py", line 645, in __setattr__
    raise ValueError(
ValueError: MaxOrder must be smaller than 3
```

Could I change the class from sextupole to a multipole ?
Could I force to change MaxOrder ?
As a work around I would guess I can add a thin multipole but it will change the number of elements in my ring ( i.e. N more elements for N magnets with multipoles).

**Comment by swhite2401** [comment]
You can normally extend the order to any value but, in __setattr__ there is a check to ensure polynomA/B and MaxOrder are consistent so setting them sequentially may fail.
You probably have to rebuild a new sextupole element from scratch with increase polynoms length and MaxOrder and replace the existing one with this new element.
This can certainly be improved....

**Reply by oscarxblanco** [reply]
alright, thank you.

**Comment by swhite2401** [comment]
In fact you can just set things in this order:
-PolynomA/B with len(PolynomA/B)>MaxOrder
-MaxOrder with MaxOrder < min (len(PolynomA, PolynomB)

This is cleaner than rebuilding the element, up to you!
